File: python/app/model/UserPerm.py
from app.model.Model import Model
from app.common.Inc import Inc

# 权限表
class UserPerm(Model):

  uid = '0'
  state_admin = '0'
  state_app = '0'
  utime = 'null'
  role = ''
  perm = ''

  # 构造函数
  def __init__(self):
    self.setPrimaryKey('uid')
    self.setSource('user_perm')

  # 更新
  def beforeUpdate(self):
    if self.utime=='null' : self.utime=Inc.date('%Y%m%d%H%M%S')

File: java/src/main/java/vip/webmis/java/model/UserPerm.java
package vip.webmis.java.model;

import com.alibaba.fastjson.JSONObject;

import vip.webmis.java.common.Inc;

/* 权限表 */
public class UserPerm extends Model {

  public String uid = "0";
  public String state_admin = "0";
  public String state_app = "0";
  public String utime = "null";
  public String role = "";
  public String perm = "";
  
  /* 构造函数 */
  public UserPerm() {
    this.setPrimaryKey("uid"); //主键
    this.setSource("user_perm"); //数据表
  }

  /* 更新 */
  public JSONObject beforeUpdate(JSONObject fields){
    if(fields.get("utime").equals("null")){
      fields.put("utime",Inc.date("yyyy-MM-dd HH:mm:ss"));
    }
    return fields;
  }

}
